src/components/bulk-email-tool/bulk-email-form/BulkEmailForm.jsx

```diff
@@ -0,0 +1,221 @@
+import TaskAlertModal from './TaskAlertModal';
+import useTimeout from '../../../utils/useTimeout';
+
+export const FORM_SUBMIT_STATES = {
+  DEFAULT: 'default',
+  PENDING: 'pending',
+  COMPLETE: 'complete',
+  COMPLETED_DEFAULT: 'completed_default',
+  ERROR: 'error',
+};
+
+export default function BulkEmailForm(props) {
+  const { courseId } = props;
+  const [subject, setSubject] = useState('');
+  const [emailFormStatus, setEmailFormStatus] = useState(FORM_SUBMIT_STATES.DEFAULT);
+  const [emailFormValidation, setEmailFormValidation] = useState({
+    // set these as true on initialization, to prevent invalid messages from prematurely showing
+    subject: true,
+    body: true,
+    recepients: true,
+  });
+  const [selectedRecipients, { add, remove }] = useCheckboxSetValues([]);
+  const [isTaskAlertOpen, openTaskAlert, closeTaskAlert] = useToggle(false);
+  const editorRef = useRef(null);
+  const resetEmailForm = useTimeout(() => {
+    setEmailFormStatus(FORM_SUBMIT_STATES.COMPLETED_DEFAULT);
+  }, 3000);
+
+  const onRecepientChange = event => {
+    if (event.target.checked) {
+      add(event.target.value);
+    } else {
+      remove(event.target.value);
+    }
+  };
+  const onInit = (event, editor) => {
+    editorRef.current = editor;
+  };
+
+  const onSubjectChange = event => setSubject(event.target.value);
+
+  const validateEmailForm = () => {
+    const subjectValid = subject.length !== 0;
+    const bodyValid = editorRef.current.getContent().length !== 0;
+    const recepientsValid = selectedRecipients.length !== 0;
+
+    setEmailFormValidation({
+      subject: subjectValid,
+      recepients: recepientsValid,
+      body: bodyValid,
+    });
+
+    return subjectValid && bodyValid && recepientsValid;
+  };
+
+  const createEmailTask = async () => {
+    const emailData = new FormData();
+    setEmailFormStatus(() => FORM_SUBMIT_STATES.PENDING);
+    const validForm = validateEmailForm();
+    if (!validForm) {
+      setEmailFormStatus(() => FORM_SUBMIT_STATES.DEFAULT);
+      return;
+    }
+    emailData.append('action', 'send');
+    emailData.append('send_to', JSON.stringify(selectedRecipients));
+    emailData.append('subject', subject);
+    emailData.append('message', editorRef.current.getContent());
+    let data;
+    try {
+      data = await postBulkEmail(emailData, courseId);
+    } catch (e) {
+      setEmailFormStatus(FORM_SUBMIT_STATES.ERROR);
+      return;
+    }
+    if (data.status === 200) {
+      setEmailFormStatus(FORM_SUBMIT_STATES.COMPLETE);
+      resetEmailForm();
+    }
+  };
+
+  return (
+    <div className="w-100 m-auto p-lg-4 py-2.5 px-5">
+      <TaskAlertModal
+        isOpen={isTaskAlertOpen}
+        alertMessage={(
+          <>
+            <p>
+              <FormattedMessage
+                id="bulk.email.task.alert.recepients"
+                defaultMessage="You are sending an email message with the subject test html to the following recipients:"
```

Comment:
Is the `test html` piece of this message supposed to be dynamic?

Reply:
I checked the bulk course email tool today and it _does_ look to be dynamic. It pulls in the `subject` as part of the message displayed in the modal. I imagine we would want to change this message to eliminate the subject being part of it or do we need to put a placeholder in the text somewhere?

Reply:
oh whoops. wow good catch. I think I must have just copy and pasted it from the original tool

Reply:
I thought this would be more of a pain to fix but it looks like this method is A OK to use.